### PR TITLE
Dao inheritance

### PR DESCRIPTION
**Pull request by kanake10**
A small PR to show how to use dao inheritance. This would be a great approach since the app will eventually have multiple daos. Using a base dao will reduce boilerplate code.

### CODE REVIEW

File: app/src/main/java/com/swayy/musicpark/presentation/NapsterApp.kt [hunk 1]
```diff
@@ -2,6 +2,16 @@ package com.swayy.musicpark.presentation
...
\ No newline at end of file
+class NapsterApp : Application() {
+    override fun onCreate() {
+        super.onCreate()
+        setupTimber()
+    }
+}
+
+private fun setupTimber() {
+    Timber.plant(Timber.DebugTree())
```
Comment (Felix-Kariuki) [inline app/src/main/java/com/swayy/musicpark/presentation/NapsterApp.kt]: 👍 Consider checking if the build being run is debug to ensure logs are visible only for debug builds not release builds ---> if (BuildConfig.DEBUG) Timber.plant(Timber.DebugTree())

File: app/src/main/java/com/swayy/musicpark/data/local/dao/BaseDao.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package com.swayy.musicpark.data.local.dao
```
Comment (Felix-Kariuki) [inline app/src/main/java/com/swayy/musicpark/data/local/dao/BaseDao.kt:1]: This is great to introduce a base dao for inheritance and reusability in the project reducing boilerplate code 👏👏 ata mimi sikuwa najua a person can do it this way learning never ends 😂

**Review by Felix-Kariuki — APPROVED** [review]

File: app/src/main/java/com/swayy/musicpark/presentation/NapsterApp.kt (hunk 1 above)
Comment (kanake10) [reply app/src/main/java/com/swayy/musicpark/presentation/NapsterApp.kt]: sure ,I will customize timber for debug and release.

Comment (kanake10) [reply app/src/main/java/com/swayy/musicpark/presentation/NapsterApp.kt]: is it okay now ? @Felix-Kariuki

File: app/src/main/java/com/swayy/musicpark/presentation/NapsterApp.kt [hunk 3]
```diff
@@ -1,17 +1,29 @@
...
 import dagger.hilt.android.HiltAndroidApp
 import timber.log.Timber
 
+
 @HiltAndroidApp
 class NapsterApp : Application() {
     override fun onCreate() {
         super.onCreate()
-        setupTimber()
+        if (BuildConfig.DEBUG) {
```
Comment (Felix-Kariuki) [inline app/src/main/java/com/swayy/musicpark/presentation/NapsterApp.kt:14]: 👍

**Review by Felix-Kariuki — APPROVED** [review]

File: app/src/main/java/com/swayy/musicpark/data/local/dao/BaseDao.kt (hunk 2 above)
Comment (kanake10) [reply app/src/main/java/com/swayy/musicpark/data/local/dao/BaseDao.kt:1]: student for life 😂

**Pull request merged**